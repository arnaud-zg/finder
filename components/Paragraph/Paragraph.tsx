import React from 'react'
import { style } from 'typestyle'
import { rem } from 'csx'
import { centerCenter } from 'csstips'

const className = style({
  fontSize: rem(1),
  margin: rem(0.5),
})

interface IProps {
  center?: boolean
  content: string
}

export const Paragraph = (props: IProps) => {
  const { center, content } = props
  return (
    <p className={`${className}${center ? ` ${style(centerCenter)}` : ''}`}>
      {content}
    </p>
  )
}
