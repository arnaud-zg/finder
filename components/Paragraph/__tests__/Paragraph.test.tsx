import React from 'react'
import renderer from 'react-test-renderer'
import { Paragraph } from '../Paragraph'

describe('Paragraph', () => {
  it('renders correctly', () => {
    const testRenderer = renderer.create(<Paragraph content="Hello World" />)
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })

  it('renders correctly with center props', () => {
    const testRenderer = renderer.create(
      <Paragraph center={true} content="Hello World" />
    )
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
})
