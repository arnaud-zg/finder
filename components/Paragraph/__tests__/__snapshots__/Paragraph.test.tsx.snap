// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph renders correctly 1`] = `
<p
  className="f2qtt8h"
>
  Hello World
</p>
`;

exports[`Paragraph renders correctly with center props 1`] = `
<p
  className="f2qtt8h f1jc9aff"
>
  Hello World
</p>
`;
